docs/excel/excel-add-ins-multiple-ranges.md

```diff
@@ -0,0 +1,208 @@
+    formulaRanges.format.fill.color = "pink";
+
+    return context.sync();
+})
+```
+
+In many scenarios in which you would need to discover the cells on which to operate (maybe most such scenarios), there sometimes won't be *any* cells with the targeted characteristic. If `getSpecialCells` doesn't find any, it throws an **ItemNotFound** error. This would divert the flow of control to a `catch` block/method, if there is one. If there isn't, the error just halts the function. There may be scenarios in which throwing the error is exactly what you want to happen when there are no cells with the targeted characteristic. But in scenarios in which it is normal, but perhaps uncommon, for there to be no matching cells; your code should check for this possibility and handle it gracefully without throwing an error. For these scenarios, use the `getSpecialCellsOrNullObject` method and test the `RangeAreas` object that is returned for nullity. The following is an example. Note about this code:
+
+- The `getSpecialCellsOrNullObject` method always returns a proxy object, so it is never `null` in the ordinary JavaScript sense. But if no matching cells are found, the `isNullObject` property of the object is set to `true`.
+- It calls `context.sync` *before* it tests the `isNullObject` property. This is a requirement with all `*OrNullObject` methods and properties.
+- You can test this code by first selecting a range that has no formula cells and running it. Then select a range that has at least one cell with a formula and run it again.
+
+```js
+Excel.run(function (context) {
+    const range = context.workbook.getSelectedRange();
+    const formulaRanges = range.getSpecialCellsOrNullObject("Formulas");
+    return context.sync()
+        .then(function() {
+            if (formulaRanges.isNullObject) {
+                console.log("No cells have formulas");
+            }
+            else {
+                formulaRanges.format.fill.color = "pink";
+            }
+        })
+        .then(context.sync);
+})
+```
+
+For simplicity, all other examples in this article will use the `getSpecialCells` method instead of  `getSpecialCellsOrNullObject`.
+
+#### Narrow the target cells with cell value types
+
+If your are passing either "Formulas" or "Constants" to `getSpecialCells` or `getSpecialCellsOrNullObject`, there is an optional second parameter, of enum type `Excel.SpecialCellValueType`, that can further narrow down the cells to target. You can use the parameter to specify that you only want cells having certain types of values. There are four basic types: "Error", "Logical" (which means boolean), "Numbers", and "Text". (The enum has other values besides these four which are discussed below.) The following is an example. About this code, note:
+
+- It will highlight all, and only, cells that have a literal number value. It will not highlight cells that display a number, but have a formula; nor will highlight any boolean, text, or error state cells. 
```

Comment:
Consider "It will highlight all, and only, cells" -> "It will only highlight cells" (Or maybe even remove "only" because you clarify the not conditions so well right after).